### ISSUE TITLE
2.9.0: H5Vol w/ HDF5 1.14.0 Fails Compile on Win

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#3649]
**Describe the bug**
HDF5 with h5Vol does not compile on Windows with:
- ADIOS 2.9.0
- HDF5 1.14.0

**To Reproduce**
See https://github.com/conda-forge/adios2-feedstock/pull/71

Errors:
```
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(40): error C2146: syntax error: missing ')' before identifier 'ndims'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(40): error C2081: 'uint': name in formal parameter list illegal
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(40): error C2061: syntax error: identifier 'ndims'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(40): error C2059: syntax error: ';'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(40): error C2059: syntax error: ')'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(42): error C2146: syntax error: missing ')' before identifier 'ndims'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(42): error C2081: 'uint': name in formal parameter list illegal
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(42): error C2061: syntax error: identifier 'ndims'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(42): error C2059: syntax error: ';'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolUtil.h(42): error C2059: syntax error: ')'
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolReadWrite.c(19): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
NMAKE : fatal error U1077: 'D:\bld\adios2-split_1685654832661\_build_env\Library\bin\cmake.exe' : return code '0x2'
```

**Expected behavior**
Compile.

**Desktop (please complete the following information):**
 - OS/Platform: Windows
 - Build: MSVC 19.29.30148.0 w/ latest CMake and NMake

**Additional context**
Needed to unblock our 2.9.0 release on conda-forge: https://github.com/conda-forge/adios2-feedstock/pull/71

The issue here has higher priority than #3648.

cc @guj

**Comment by ax3l** [comment]
```
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolReadWrite.c(19): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
```
aka
```C++
#include <unistd.h> // sleep
```

is not portable. You can try
https://stackoverflow.com/a/8764595

or something ISO C++11 and newer such as https://en.cppreference.com/w/cpp/thread/sleep_for

**Comment by ax3l** [comment]
The issues here I do not immediately understand:
https://github.com/ornladios/ADIOS2/blob/v2.9.0/source/h5vol/H5VolUtil.h#L40-L42

Maybe you need to write https://en.cppreference.com/w/cpp/types/size_t and import `<cstddef>`.

**Comment by ax3l** [comment]
```
D:\bld\adios2-split_1685654832661\work\source\h5vol\H5VolReadWrite.c(19): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
```
aka
```C++
#include <unistd.h> // sleep
```

is not portable. You can try
https://stackoverflow.com/a/8764595

or something ISO C++11 and newer such as https://en.cppreference.com/w/cpp/thread/sleep_for

**Comment by pnorbert** [comment]
@vicentebolea Can we please have HDF5  on for the win tests?

**Comment by ax3l** [comment]
Fixed via #3652.

Follow-up task (CI coverage) in #3662
